src/main/java/pl/zimi/zadanka/Zad5.java

```diff
@@ -0,0 +1,62 @@
+package pl.zimi.zadanka;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+
+/**
+ * source:
+ * https://leetcode.com/problems/climbing-stairs/
+ */
+
+public class Zad5 {
+
+    // rozwiązanie poprawne ale wolne
+    public static int climbStairsTreeStyle(int key, int currentValue, int carry) {
+        if (currentValue == key) {
+            return ++carry;
+        }
+        if (currentValue > key) {
+            return 0;
+        } else {
+            return climbStairsTreeStyle(key, currentValue + 1, carry)
+                    + climbStairsTreeStyle(key, currentValue + 2, carry);
+        }
+    }
+
+    // rozwiązanie poprawne ale za wolne
+    public static int climbStairs(int key) {
```

Comment:
popraw nazwę "key" co biznesowo znaczy ta wartość

Reply:
:D biznesowo to nic nie znaczy bo nie zarabia pieniędzy :)
tam gdzie "key" był uzywany w kontekście zmiennej ulegającej zmianie w trakcie kolejny rekurencji przemianowałem zmienną na stairNumber. W miejscach gdzie kontekst zmiennej key  wskazywał, że jej wartość będzie służyła do porównywania ze zmienną ulegającą zmianie w trakcie wykonywania metody przemianowałem key na lastStairNumber